pkg/chart/chart.go

```diff
@@ -323,24 +331,32 @@ func (c *HelmRepoClient) Init(appRepo *appRepov1.AppRepository, caCertSecret *co
 
 // GetChart retrieves and loads a Chart from a registry in both
 // v2 and v3 formats.
-func (c *HelmRepoClient) GetChart(details *Details, repoURL string) (*helm3chart.Chart, error) {
+func (c *HelmRepoClient) GetChart(details *Details, repoURL string) (*chart.Chart, error) {
 	if c.netClient == nil {
 		return nil, fmt.Errorf("unable to retrieve chart, Init should be called first")
 	}
-	var chart *helm3chart.Chart
-	indexURL := strings.TrimSuffix(strings.TrimSpace(repoURL), "/") + "/index.yaml"
-	repoIndex, err := fetchRepoIndex(&c.netClient, indexURL)
-	if err != nil {
-		return nil, err
-	}
-
-	chartURL, err := findChartInRepoIndex(repoIndex, indexURL, details.ChartName, details.Version)
-	if err != nil {
-		return nil, err
+	var chartURL string
+	var err error
+	if details.TarballURL != "" {
+		chartURL, err = resolveChartURL(repoURL, details.TarballURL)
+		if err != nil {
+			return nil, err
+		}
+	} else {
```

Comment:
Excellent, so this is the main change: don't fetch the index if we already have the tarball URL.

What are the kubeops cases where details.TarballURL is "", out of interest?

Reply:
It is just the same scenario as in the Helm-direct plugin: creating and upgrading releases. Since these operations are no longer used, it is safe to delete the expensive if-branch once we deprecate this service.

https://github.com/kubeapps/kubeapps/blob/f9f24694b1fb5717ba10d1a6101cbce6d5aad41e/cmd/kubeops/internal/handler/handler.go#L188

https://github.com/kubeapps/kubeapps/blob/f9f24694b1fb5717ba10d1a6101cbce6d5aad41e/cmd/kubeops/internal/handler/handler.go#L241